doc/KUKSA.val_data_broker/authorization.md

```diff
@@ -0,0 +1,505 @@
+| `provide:value` | Allow client to provide the value (but not respond to actuation requests) for matching signals. (includes `read`) |
+| `provide:actuation` | Allow client to respond to actuation requests (but not provide the value) for matching signals. (includes `read`) |
+
+| Subactions for `edit` | Description                     |
+|-----------------------|---------------------------------|
+| `edit:field:FIELD` | Allow client to edit metadata field FIELD for matching signals. (includes `read`) |
+
+
+#### Add "deny" scopes to limit other scopes
+| Scope                 | Description                 |
+|-----------------------|-----------------------------|
+|`!<ACTION>:<PATH>`     | Deny ACTION for PATH        |
+
+All "deny" scopes have priority over any "allow" scope.
+
+Examples of KUKSA.VAL scope strings:
+
+| Scope string             | Access                                        |
+|--------------------------|-----------------------------------------------|
+|`read:Vehicle`            | Client can read everything under path `Vehicle` (except what is denied below) |
+|`!read:field:sensitive_field` | Client is _not_ allowed to read `sensitive_field` |
+|`!read:Vehicle.Sensitive.Path` | Client is _not_ allowed to read anything under `Vehicle.Sensitive.Path` |
+
+
+#### Add "tag" to scope as alternative to path
+Another possible extension is to allow something other than paths to identify a (group of)
+signal(s).
+
+Once scenario is that categorization information (tags) has been added to the VSS `.vspec` files.
+This information would be available as metadata for VSS entries and could be used to
+identify / select signals without the need to know their paths.
+
+This isn't something that is supported today (and might never be), but it's included here to
+demonstrate the extensibility of the scope syntax as shown here.
+
+| Scope                                   | Description                              |
+|-----------------------------------------|------------------------------------------|
+|`<ACTION>:field:<FIELD>:tag:<TAG>`  | Allow ACTION for field FIELD for signals tagged with TAG |
+|`!<ACTION>:field:<FIELD>:tag:<TAG>` | Deny ACTION for field FIELD for signals tagged with TAG  |
+|`<ACTION>:tag:<TAG>`                     | Allow ACTION for signals tagged with TAG |
+|`!<ACTION>:tag:<TAG>`                    | Deny ACTION for signals tagged with TAG  |
+
+Example:
+
+```
+"!read:tag:restricted !actuate:tag:restricted read:tag:low_impact actuate:tag:low_impact"
+```
+
+#### Add "vin" to scope to limit access per vehicle
+As described in [Limit access to resources](#limit-access-to-resources), it's a good idea
+to limit what resources an access token is authorized to access. The `"aud"` claim seems
+to be a good way to do this.
+
+A possible addition to this would be to be able to limit access to certain VIN:s by also
+introducing a `"vin:<VIN>"` scope.
+
+```yaml
+{
+  ...
+  "scope": "... vin:5GZCZ43D13S812715"
```

Comment:
Not so sure, this is needed. So if combined with other read/etc scopes would it _restrict_ those further to only one VIN? Or does adding this scope imply "for this Vehicle "do whatever"

I feel if we define how aud works, how to use "aud" to not limit to a VIN, and how an "access whatever" scope (or then non-scope? See comment above) works this is not adding any clarity.

If we see this as an alternative, I like the `aud` better

Reply:
> Not so sure, this is needed.

Me neither. Is one more "possible additions" meant to evaluate the extensibility of the design.

> So if combined with other read/etc scopes would it restrict those further to only one VIN?

Yes, that's the idea. Should be stated more clearly.

> If we see this as an alternative, I like the aud better

It's just an additional proposal and it's something that would be added in addition to the `"aud"` claim (as stated in the description of this scope addition).

**Example use case**

Let's say that a resource server has been configured to accept the following audiences in the access token (as proposed in a comment above):
```
[authorization]
accepted_audiences = ["asd8f98cv789", "234098vcxc"]
```

If 20 resource servers (vehicles) are all configured in this way, it would mean that an access token issued for one of those would be valid for all. If a resource owner wants to issue an access token valid for just one of them, there is no way to do this using the `"aud"` claim. It would be if all resource servers had at least one unique `accepted_audiences` entry.

The idea of adding it as a scope as well, is to allow the resource owner to restrict the access token to just one of those vehicles (even though they are all hard-coded to accept the same `"aud"`).

```
"scope": "... vin:5GZCZ43D13S812715"
```

This would act in addition to all the other scope:s specified. It wouldn't include any specific access, i.e. `read`. It would just mean that if `vin:5GZCZ43D13S812715` doesn't match the VIN of the vehicle, nothing else present in the scope would be allowed either.

But perhaps a better design would be to just make sure that all resource servers have at least one unique `accepted_audiences`. But as you said somewhere, this might not always be possible in a mass production scenario?

Reply:
See comment above, maybe we can do two things
 - Allow a list of audiences to be configured when starting databroker
 - Without specifying any, just define a default audience, i.e. `kuksa.val`(not married to the string, could also be `DEADBEEF`or `AFFE`...? , which we can also use in example tokens

Reply:
Yes, I don't immediately see an issue with having a default audience.

Reply:
See comment above wrt to audiences